tests/test_interrupted_sync.py

```diff
@@ -0,0 +1,279 @@
+                self.assertTrue(self.is_expected_date_format(first_bookmark_value))
+                self.assertIsNotNone(resuming_bookmark_value)
+                self.assertTrue(self.is_expected_date_format(resuming_bookmark_value))
+
+                # verify the resuming bookmark is greater than 1st sync bookmark
+                # This is the expected behaviour for shopify as they are using date windowing
+                # TDL-17096 : Resuming bookmark value is getting assigned from execution time
+                # rather than the actual bookmark time for some streams.
+                # TODO this is not the case for the transactions stream, they are equal.
+                if stream == 'transactions':
+                    self.assertEqual(resuming_bookmark_value, first_bookmark_value)
+                else:
+                    self.assertGreater(resuming_bookmark_value, first_bookmark_value)
+
+                # verify oldest record from resuming sync respects bookmark from previous sync
+                if stream in new_state['bookmarks'].keys() and resuming_sync_messages:
+                    self.assertEqual(actual_oldest_resuming_replication_date,
+                                     self.parse_date(simulated_bookmark_value),
+                                     msg="Oldest resuming sync record not respecting bookmark")
+
+                    # all interrupted recs are in full recs, interrupted rec counts verified
+                    first_sync_records_after_bookmark = [
+                        record for record in first_sync_messages
+                        if self.parse_date(record[replication_key]) >=
+                        self.parse_date(simulated_bookmark_value)]
+                    # remove any records that got added after the first sync
+                    filtered_resuming_records = [
+                        record for record in resuming_sync_messages
+                        if self.parse_date(record[replication_key]) <=
+                        youngest_first_sync_date]
+
+                    self.assertEqual(first_sync_records_after_bookmark,
+                                     filtered_resuming_records,
+                                     msg="Incorrect data in the resuming sync")
+
+                for record in resuming_sync_messages:
+                    replication_key_value = record.get(replication_key)
+                    # this assertion is only for completed and interrupted streams
+                    if stream in new_state['bookmarks'].keys():
+                        # verify 2nd sync rep key value is greater or equal to 1st sync bookmarks
+                        self.assertGreaterEqual(replication_key_value, simulated_bookmark_value,
+                                                msg=("Resuming sync records do not respect the"
+                                                     " previous bookmark")
+                                                )
```

Comment:
Another way we can shorten the line here is like this
```suggestion
                        self.assertGreaterEqual(
                            replication_key_value, 
                            simulated_bookmark_value,
                            msg="Resuming sync records do not respect the previous bookmark"
                        )
```

I see a line length of 94 in emacs despite Github's wrapping

Reply:
I prefer having both values in the assert on the same line when possible so the tap tester failure message instantly gives context as to what is being compared rather than having to look at the code to start the debug.

Reply:
> I prefer having both values in the assert on the same line when possible

That's fine, but that doesn't have to affect `msg`.
```suggestion
                        self.assertGreaterEqual(
                            replication_key_value, simulated_bookmark_value,
                            msg="Resuming sync records do not respect the previous bookmark"
                        )
```

> the tap tester failure message instantly gives context as to what is being compared

Could you elaborate on this? I can't write up a failing test that shows this

```python
python -V; nosetests -V; cat test.py; nosetests test.py
Python 3.9.16
nosetests version 1.3.7
import unittest

class TestFoo(unittest.TestCase):

    maxdiff = None

    def test_one(self):
        self.assertGreaterEqual(
            1, 2,
            msg="this is message 1"
        )

    def test_two(self):
        self.assertGreaterEqual(
            1,
            2,
            msg="this is message 2"
        )
FF
======================================================================
FAIL: test_one (test.TestFoo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alu/tmp/test.py", line 8, in test_one
    self.assertGreaterEqual(
AssertionError: 1 not greater than or equal to 2 : this is message 1

======================================================================
FAIL: test_two (test.TestFoo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alu/tmp/test.py", line 14, in test_two
    self.assertGreaterEqual(
AssertionError: 1 not greater than or equal to 2 : this is message 2

----------------------------------------------------------------------
Ran 2 tests in 0.001s

FAILED (failures=2)
```

Reply:
if you put the values on the same line as the assert then they show up in the failure message and you know what is being compared.  In the failure messages shown here we have no context as to what "1" and "2" actually represent so we have to go look at the code.

Reply:
> if you put the values on the same line as the assert then they show up in the failure message and you know what is being compared

I am on the same page as you now. I was confused because when you say "the assert", I'm thinking of 

```python
        self.assertGreaterEqual(
            1,
            2,
            msg="this is message 2"
        )
```

But I think you're talking about

```python
        self.assertGreaterEqual(
```

> In the failure messages shown here we have no context as to what "1" and "2" actually represent so we have to go look at the code

I would love to chat about how this fits into your workflow when looking at a test failure. In my opinion we should not be forcing ourselves to format code a certain way just to change how the stack trace is printed.

This feels more like a misuse of `nose` as a tool. Here's the same example from above but with a third test where we use the Python `assert` function instead of the `unittest.TestCase.assert*` functions.

```python
$ python -V; nosetests -V; cat test_foo.py; nosetests --detailed-errors test_foo.py
Python 3.9.16
nosetests version 1.3.7
import unittest

class TestFoo(unittest.TestCase):

    def test_one(self):
        self.assertGreaterEqual(
            1, 2,
            msg="this is message 1"
        )

    def test_two(self):
        expected = 1
        actual = 2
        assert expected >= actual, "this is message 2"

    def test_three(self):
        expected = 1
        actual = 2
        assert \
            expected >= actual, \
            "this is message 3"
FFF
======================================================================
FAIL: test_one (test_foo.TestFoo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alu/tmp/test_foo.py", line 6, in test_one
    self.assertGreaterEqual(
AssertionError: 1 not greater than or equal to 2 : this is message 1
    """Fail immediately, with the given message."""
>>  raise self.failureException('1 not greater than or equal to 2 : this is message 1')


======================================================================
FAIL: test_three (test_foo.TestFoo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alu/tmp/test_foo.py", line 19, in test_three
    assert \
AssertionError: this is message 3
    1 = 1
    2 = 2
>>  assert \
        1 >= 2, \
        "this is message 3"


======================================================================
FAIL: test_two (test_foo.TestFoo)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/alu/tmp/test_foo.py", line 14, in test_two
    assert expected >= actual, "this is message 2"
AssertionError: this is message 2
    1 = 1
    2 = 2
>>  assert 1 >= 2, "this is message 2"


----------------------------------------------------------------------
Ran 3 tests in 0.031s

FAILED (failures=3)
```

I don't know enough about the relationship between `nosetests` and `python` and `unittests` - maybe the author was super opinionated and just doesn't support `unittests`. But by using `assert`, we get the context you want and without having to format the code in any particular way.

Reply:
I'm sorry, I'm wrong here. This isn't about `nose` or `pytests` because we are talking about the integration tests. So the runner is `tap-tester`